node/network/availability-distribution/src/requester/mod.rs

```diff
@@ -98,36 +98,36 @@ impl Requester {
 		tracing::trace!(target: LOG_TARGET, ?update, "Update fetching heads");
 		let ActiveLeavesUpdate { activated, deactivated } = update;
 		// Stale leaves happen after a reversion - we don't want to re-run availability there.
-		let activated = activated.and_then(|h| match h.status {
+		if let Some(leaf) = activated.and_then(|h| match h.status {
```

Comment:
`activated.filter(|head| head.status == LeafStatus::Fresh)` also works and is shorter